onnxruntime/python/tools/transformers/optimizer.py

```diff
@@ -174,13 +181,41 @@ def optimize_by_onnxruntime(
         else:
             providers.append("CUDAExecutionProvider")
 
-    onnxruntime.InferenceSession(onnx_model_path, sess_options, providers=providers, **kwargs)
+    if isinstance(onnx_model, str):
+        onnxruntime.InferenceSession(onnx_model, sess_options, providers=providers, **kwargs)
+    elif isinstance(onnx_model, ModelProto):
+        _load_infer_session_from_modelproto(onnx_model, save_as_external_data, sess_options, providers, kwargs)
 
     assert os.path.exists(optimized_model_path) and os.path.isfile(optimized_model_path)
     logger.debug("Save optimized model by onnxruntime to %s", optimized_model_path)
     return optimized_model_path
 
 
+def _load_infer_session_from_modelproto(model, has_external_data, sess_options, providers, kwargs) -> None:
```

Comment:
Prefer adding docstrings to any new functions: https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html